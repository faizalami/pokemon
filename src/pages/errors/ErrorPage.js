function ErrorPage ({ code, message }) {
  return (
    <>
      <div>
        { code }
      </div>
      <div>
        { message }
      </div>
    </>
  )
}

export default ErrorPage;
