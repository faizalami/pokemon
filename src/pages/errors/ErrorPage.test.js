import { render } from '@testing-library/react';
import ErrorPage from './ErrorPage';

test('Error page matches snapshot', () => {
  expect(render(<ErrorPage code={404} message="Not Found"/>)).toMatchSnapshot();
});
