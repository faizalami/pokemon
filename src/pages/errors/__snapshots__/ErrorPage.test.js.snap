// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
}

.emotion-2 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <picture>
          <source
            data-testid="image-webp"
            srcset="/assets/img/webp/surprised-pikachu-128.webp"
            type="image/webp"
          />
          <source
            data-testid="image-png"
            srcset="/assets/img/png/surprised-pikachu-128.png"
            type="image/png"
          />
          <img
            class="emotion-1"
            data-testid="image-regular"
            height="128"
            loading="eager"
            src="/assets/img/png/surprised-pikachu-128.png"
            width="128"
          />
        </picture>
        <h1
          class="emotion-2"
        >
          404
        </h1>
        <p
          class="emotion-2"
        >
          Not Found
        </p>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
}

.emotion-2 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

<div>
    <div
      class="emotion-0"
    >
      <picture>
        <source
          data-testid="image-webp"
          srcset="/assets/img/webp/surprised-pikachu-128.webp"
          type="image/webp"
        />
        <source
          data-testid="image-png"
          srcset="/assets/img/png/surprised-pikachu-128.png"
          type="image/png"
        />
        <img
          class="emotion-1"
          data-testid="image-regular"
          height="128"
          loading="eager"
          src="/assets/img/png/surprised-pikachu-128.png"
          width="128"
        />
      </picture>
      <h1
        class="emotion-2"
      >
        404
      </h1>
      <p
        class="emotion-2"
      >
        Not Found
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
